cmake/projects/GTest/hunter.cmake

```diff
@@ -183,7 +183,20 @@ hunter_add_version(
     4fe083a96d7597f7dce6f453dca01e1d94a1e45b
 )
 
-if(HUNTER_GTest_VERSION VERSION_LESS 1.8.0)
+hunter_add_version(
+    PACKAGE_NAME
+    GTest
+    VERSION
+    1.8.0-hunter-p8
+    URL
+    "https://github.com/hunter-packages/googletest/archive/1.8.0-hunter-p8.tar.gz"
+    SHA1
+    ab0f563b3bd80a1721cb93df034a86ec
+)
+
+# The regexp case is used when the package is added as GIT_SUBMODULE.
+# In this case we assume that the version is later than 1.8.0.
+if(HUNTER_GTest_VERSION VERSION_LESS 1.8.0 AND NOT MATCHES "^[a-z0-9]{40}$")
```

Comment:
I don't think it's correct condition. Try:
```cmake
set(A "0.0")
if(A VERSION_LESS 1.0 AND NOT MATCHES "^[a-z0-9]{40}$")
  message("A < 1.0")
  message("A is not SHA1")
endif()
```

I guess CMake interpret it as `"NOT" MATCHES "^[a-z0-9]{40}$"`which is `FALSE`.